### ISSUE TITLE
Unexpected bad performance 

### ISSUE THREAD

**Issue by tzlils** [bevyengine/bevy#3687]
## Bevy version
0.6

## Operating system & version
Windows 10.0.19043 Build 19043

## What you did
Compiled and ran the [many_cubes](https://github.com/bevyengine/bevy/blob/main/examples/3d/many_cubes.rs) example with opt-level=3 in debug mode and release mode

## What you expected to happen
The 3D scene is rendered at my monitor's refresh rate (60.056)

## What actually happened
Surprisingly low FPS

## Additional information
![image](https://user-images.githubusercontent.com/35463279/149642492-5e6ade90-09af-4196-8967-d903502c75c1.png)
`bevy_render::renderer: AdapterInfo { name: "Intel(R) UHD Graphics", vendor: 32902, device: 35414, device_type:
IntegratedGpu, backend: Vulkan }`

the FPS was about the same in release and in debug mode, it was not affected by turning off vsync

**Comment by alice-i-cecile** [comment]
Could you check your FPS on some other examples? 2D would be nice to compare against, as would the web examples https://bevyengine.org/examples/ (bevymark for example has an FPS counter).

**Comment by tzlils** [comment]
![image](https://user-images.githubusercontent.com/35463279/149643005-1bae651e-9405-4349-8b01-fa5e1ac742a9.png)
seeing stable 60 fps on bevymark,
drops to 20 FPS at around 52k birds.

compiling and running the [many_sprites](https://github.com/bevyengine/bevy/blob/main/examples/2d/many_sprites.rs) example on release with opt-level=3 and getting 24 FPS average

**Comment by GioPat** [comment]
I'm having a similar (using alien cake addict example) issue: https://github.com/bevyengine/bevy/issues/3027#issuecomment-1012844856

I'm keen to do further tries if you need!

**Comment by superdump** [comment]
Could you all try with `WGPU_OPTIONS_PRIO=compatibility` to see if that makes a difference? I'm trying to find cases where https://github.com/bevyengine/bevy/issues/3686 has observable practical impacts and this could be one.

**Comment by GioPat** [comment]
Tried running with this command, setting WGPU_OPTIONS_PRIO as env variable:

`set WGPU_OPTIONS_PRIO=compatibility&& cargo run --example alien_cake_addict --release`

I have an improvement of ~10 FPS: Now running at 44/45 FPS

**Comment by superdump** [comment]
Perfect, thanks. Others experiencing unexpectedly poor performance are welcome to test too as it is interesting to hear if all cases are due to #3686 or if there are other issues to be addressed.

**Comment by GioPat** [comment]
Also run the bevymark test as advised by @alice-i-cecile these are the results:

**Debug**
Hitting 15FPS @ 15000 Birds 🐦 

**Release with no WGPU_OPTIONS_PRIO**
Hitting 10 FPS @ 78000 Birds 🐦 

**Release with WGPU_OPTIONS_PRIO=compatibility **
Hitting 10 FPS @ 78000 Birds 🐦 (same as the previous test)

**Comment by tzlils** [comment]
heres trace_chrome for many_cubes (without WGPU_OPTIONS_PRIO)
[trace-1642343779.txt](https://github.com/bevyengine/bevy/files/7877149/trace-1642343779.txt)
not getting any more FPS with WGPU_OPTIONS_PRIO=compatibility in many_cubes

**Comment by GioPat** [comment]
Even though @superdump has closed a PR which is slighlty improving the performances (at least in my cases) 🙏 
Don't you think the performances are still low?
I'm not a 3d developer but if you need I could try to look into something following your guidance.

**Comment by superdump** [comment]
Yes, I think at least establishing whether what you see is expected or unexpected for your GPU would be good. I see in the original report that the GPU is Intel UHD Graphics but I don't know how to look up the vendor + device number to find the specific model in order to look up its relative performance. How many birds can you draw in bevymark at 60 fps?

**Comment by superdump** [comment]
The device number 35414 printed as hex is 8a56 which corresponds to an 11th gen ice lake Intel UHD G1 integrated GPU with 32 execution units and a maximum clock speed of 1.1GHz. https://www.notebookcheck.net/Intel-UHD-Graphics-G1-Ice-Lake-32-EU-Laptop-GPU.422873.0.html It seems to be the slowest integrated GPU configuration in the 11th generation. I'm not sure what benchmarks on notebookcheck are a good point of reference. Some are about half the performance of an M1 Max, others are about 1/15th. I think last I checked I could get around 120k birds at 60fps on an M1 Max, at least somewhere around there. So maybe around 10k at 60fps could be expected, or maybe you should be able to expect 60k. I don't know.

Another approach would be to say that some other engine can do Nk at 60fps in a way that is comparable (i.e. moving sprites).

**Comment by GioPat** [comment]
I'm going to provide the info also for my GPU as soon as possible.
@superdump I think the approach to test moving sprites with other engines is the way to go, we need to select which engines to use for this benchmark, for example I'd exclude Godot since it is using OpenGL (so I expect it will have poor performances).

Do you have any advise for the engines to be used?

**Comment by james7132** [comment]
There's been multiple performance improvements that have been merged since this issue has been filed. Could you try this again?
